File: app/src/main/java/com/phlox/tvwebbrowser/activity/main/dialogs/FavoriteEditorDialog.java
package com.phlox.tvwebbrowser.activity.main.dialogs;

import android.app.Dialog;
import android.content.Context;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.TextView;

import com.phlox.tvwebbrowser.R;
import com.phlox.tvwebbrowser.model.FavoriteItem;

/**
 * Created by PDT on 13.09.2016.
 */
public class FavoriteEditorDialog extends Dialog {
    private final FavoriteItem item;
    private final TextView tvTitle;
    private final TextView tvUrl;
    private final EditText etTitle;
    private final EditText etUrl;
    private final ImageButton ibTitle;
    private final ImageButton ibUrl;
    private final Button btnDone;
    private final Button btnCancel;
    private Callback callback;

    public interface Callback {
        void onDone(FavoriteItem item);
    }

    public FavoriteEditorDialog(Context context, final Callback callback, FavoriteItem item) {
        super(context);
        this.callback = callback;
        this.item = item;
        setCancelable(true);
        setTitle(item.id == 0 ? R.string.new_bookmark : R.string.edit);
        setContentView(R.layout.dialog_new_favorite_item);
        tvTitle = (TextView) findViewById(R.id.tvTitle);
        tvUrl = (TextView) findViewById(R.id.tvUrl);
        etTitle = (EditText) findViewById(R.id.etTitle);
        etUrl = (EditText) findViewById(R.id.etUrl);
        ibTitle = (ImageButton) findViewById(R.id.ibTitle);
        ibUrl = (ImageButton) findViewById(R.id.ibUrl);
        btnDone = (Button) findViewById(R.id.btnDone);
        btnCancel = (Button) findViewById(R.id.btnCancel);

        ibTitle.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                ibTitle.setVisibility(View.GONE);
                tvTitle.setVisibility(View.GONE);
                etTitle.setVisibility(View.VISIBLE);
                etTitle.requestFocus();
            }
        });

        ibUrl.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                ibUrl.setVisibility(View.GONE);
                tvUrl.setVisibility(View.GONE);
                etUrl.setVisibility(View.VISIBLE);
                etUrl.requestFocus();
            }
        });

        btnDone.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                FavoriteEditorDialog.this.item.title = etTitle.getText().toString();
                FavoriteEditorDialog.this.item.url = etUrl.getText().toString();
                callback.onDone(FavoriteEditorDialog.this.item);
                dismiss();
            }
        });
        btnCancel.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                dismiss();
            }
        });

        tvTitle.setText(item.title);
        etTitle.setText(item.title);
        tvUrl.setText(item.url);
        etUrl.setText(item.url);
    }
}

File: app/src/main/java/com/phlox/tvwebbrowser/activity/main/dialogs/favorites/FavoriteEditorDialog.kt
package com.phlox.tvwebbrowser.activity.main.dialogs.favorites

import android.app.Dialog
import android.content.Context
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.ImageButton
import android.widget.TextView

import com.phlox.tvwebbrowser.R
import com.phlox.tvwebbrowser.model.FavoriteItem

/**
 * Created by PDT on 13.09.2016.
 */
class FavoriteEditorDialog(context: Context, private val callback: Callback, private val item: FavoriteItem) : Dialog(context) {
    private val tvTitle: TextView
    private val tvUrl: TextView
    private val etTitle: EditText
    private val etUrl: EditText
    private val ibTitle: ImageButton
    private val ibUrl: ImageButton
    private val btnDone: Button
    private val btnCancel: Button

    interface Callback {
        fun onDone(item: FavoriteItem)
    }

    init {
        setCancelable(true)
        setTitle(if (item.id == 0L) R.string.new_bookmark else R.string.edit)
        setContentView(R.layout.dialog_new_favorite_item)
        tvTitle = findViewById<View>(R.id.tvTitle) as TextView
        tvUrl = findViewById<View>(R.id.tvUrl) as TextView
        etTitle = findViewById<View>(R.id.etTitle) as EditText
        etUrl = findViewById<View>(R.id.etUrl) as EditText
        ibTitle = findViewById<View>(R.id.ibTitle) as ImageButton
        ibUrl = findViewById<View>(R.id.ibUrl) as ImageButton
        btnDone = findViewById<View>(R.id.btnDone) as Button
        btnCancel = findViewById<View>(R.id.btnCancel) as Button

        ibTitle.setOnClickListener {
            ibTitle.visibility = View.GONE
            tvTitle.visibility = View.GONE
            etTitle.visibility = View.VISIBLE
            etTitle.requestFocus()
        }

        ibUrl.setOnClickListener {
            ibUrl.visibility = View.GONE
            tvUrl.visibility = View.GONE
            etUrl.visibility = View.VISIBLE
            etUrl.requestFocus()
        }

        btnDone.setOnClickListener {
            this@FavoriteEditorDialog.item.title = etTitle.text.toString()
            this@FavoriteEditorDialog.item.url = etUrl.text.toString()
            callback.onDone(this@FavoriteEditorDialog.item)
            dismiss()
        }
        btnCancel.setOnClickListener { dismiss() }

        tvTitle.text = item.title
        etTitle.setText(item.title)
        tvUrl.text = item.url
        etUrl.setText(item.url)
    }
}
